dockerfiles/sentieon/sentieon_ReadWriter.sh

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env bash
+
+# *******************************************
+# Merge multiple BAM files.
+# Multiple regions can be speficied to use for the merge.
+# RG IDs from the multiple files will be made unique
+# by adding a random 5 letter string.
+# *******************************************
+
+# *******************************************
+# FUNCTIONS
+# *******************************************
+display_usage() {
+    echo "Usage: $0 -r genome_reference_fasta -i input_file_bam [-i input_file_bam ...] [-l region ...]"
+    exit 1
+}
+
+generate_random_string() {
+    echo "$(cat /dev/urandom | tr -dc 'a-zA-Z' | fold -w 5 | head -n 1)"
+}
+
+process_bam_header() {
+    # Variable to store replacement @RG args
+    replace_rg_args=""
+
+    # Extract @RG lines from the header of the input BAM
+    input_bam="$1"
+    rg_lines=$(samtools view -H "$input_bam" | grep "^@RG")
+
+    # Loop through @RG lines and modify ID field
+    # Create --replace_rg arguments
+    while IFS= read -r rg_line; do
+        orig_rg_id=$(echo "$rg_line" | awk -F'\t' '/ID:/ {print $2}' | sed "s/ID://")
+        new_rg_id="${orig_rg_id}-$(generate_random_string)"
+        new_rg_line=$(echo "$rg_line" | cut -f 2- | sed "s/${orig_rg_id}/${new_rg_id}/")
+        formatted_new_rg_line=$(echo -e "$new_rg_line" | sed 's/\t/\\t/g')
+        replace_rg_args+=" --replace_rg ${orig_rg_id}='${formatted_new_rg_line}' "
```

Comment:
I think adding a comment for each line would help to understand what we are doing here